STMicroelectronics/STM32L4_L4+/samples/sprinkler_v1i1/sprinkler_v1i1_interface.c

```diff
@@ -129,5 +129,21 @@ az_result publish_interface(const az_ulib_ipc_table* const table)
 
 az_result unpublish_interface(const az_ulib_ipc_table* const table)
 {
-  return table->unpublish(&SPRINKLER_1_DESCRIPTOR, AZ_ULIB_NO_WAIT);
+  /**
+   * @note: It is the user's responsibility to halt any processes that should not persist
+   *        beyond package uninstall.
+   */
+  AZ_ULIB_TRY
+  {
+    // stop watering
+    sprinkler_1_stop_model_in stop_model_in = { 0 };
+    stop_model_in.area = 0;
+    AZ_ULIB_THROW_IF_AZ_ERROR(sprinkler_1_stop_concrete((az_ulib_model_in)&stop_model_in, NULL));
```

Comment:
To keep the isolation between the interface and the buisiness logic, lets create an end funtion in the "sprinkler_v1i1.c", something like:
```
az_result sprinkler_v1i1_end(void)
{
    return sprinkler_v1i1_stop(0);
}
```
and call it from here. Somethig like:
```
    // End sprinkler operations.
    AZ_ULIB_THROW_IF_AZ_ERROR(sprinkler_v1i1_end());
```